File: cal.c
#include <stdio.h>
#include <stdlib.h>


int main(){
int *a,*operacion;
printf("=================CALCULADORA============== \n");
printf("Operaciones disponibles \n");
printf("* 1- Suma \n");
printf("* 2- Resta \n");
printf("* 3- Multiplicacion \n");
printf("* 4- Divicion \n");
int operaciones;
scanf("%d",&operaciones);

int b;
a=(int *)malloc(b * sizeof(int));
operacion=(int *)malloc(sizeof(int));

if(operaciones==1){
	printf("===========SUMA===========");

printf("ingrese n numeros :  \n ");
scanf("%d",&b);
for (int c=1;c<=b;c++){
  
printf("ingrese el numero %d :  \n   ", c );
scanf("%d",&a[c]);

*operacion=*operacion+a[c];
}

printf("la suma es de: %d:  \n ",*operacion);
}
else if(operaciones==2){
printf("===========RESTA===========");

printf("ingrese n numeros :  \n ");
scanf("%d",&b);

for (int c=1;c<=b;c++){
   

printf("ingrese el numero %d :  \n   ", c );
scanf("%d",&a[c]);

*operacion=(*operacion)-a[c];
}

printf("la suma es de: %d:  \n ",*operacion);

}
 else if (operaciones==3)
 {
 	printf("===========MULTIPLICACION===========");

printf("ingrese n numeros :  \n ");
scanf("%d",&b);
*operacion=1;
for (int c=1;c<=b;c++){
   

printf("ingrese el numero %d :  \n   ", c );
scanf("%d",&a[c]);

*operacion=*operacion*a[c];
}

printf("la suma es de: %d:  \n ",*operacion);
 }
 else if (operaciones==4)
 {
 	printf("===========DIVISION===========");

printf("ingrese n numeros :  \n ");
*operacion=1;
scanf("%d",&b);
for (int c=0;c<b;c++){
   

printf("ingrese el numero %d :  \n   ", c );
scanf("%d",&a[c]);
*operacion=a[c]/(*operacion);
*operacion=a[c];	
printf("%d",*operacion);
}

printf("la suma es de: %d:  \n ",*operacion);
 }
 else{
 	printf("=========Opcion no encontrada========\n");
 	printf("=========FIN======\n");
 }

}

File: cal.py
import sys
print("===============CALCULADORA=============")

operacion = 0


def opciones():
    print("OPCIONES")
    print("1- SUMA")
    print("2- RESTA")
    print("3- MULTIPLICACION")
    print("4- DIVISION")
    print("5- SALIR")


def principal():
    opciones()
    acumulador = 0
    numero = 0
    cantiN = 0
    operacion = int(input())
    print("Cantidad de Numeros")
    cantiN = int(input())

    if operacion == 1:
        print("==SUMA==")
        for x in range(cantiN):
            print("Ingrese numero", x + 1)
            numero = int(input())
            acumulador = acumulador + numero
            print(">>>>>", acumulador)
        print("Total = ", acumulador)
    elif operacion == 2:
        print("==RESTA==")
        for x in range(cantiN):
            print("Ingrese numero", x + 1)
            numero = int(input())
            acumulador = -acumulador - numero
            print(">>>>>>", acumulador)
        print("Total = ", acumulador)
    elif operacion == 3:
        print("==MUlTIPLICACION==")
        acumulador = 1
        for x in range(cantiN):
            print("Ingrese numero", x + 1)
            numero = int(input())
            acumulador = acumulador * numero
            print(">>>>>>", acumulador)
        print("Total =", acumulador)
    elif operacion == 4:
        print("==DIVISION==")

        for x in range(cantiN):
            print("Ingrese numero", x + 1)
            numero = int(input())
            if x == 0:
                acumulador = numero
                print(">>>>>>>>>>>>>>", acumulador)
            elif x >= 1:
                acumulador = acumulador / numero
                print(">>>>>>>>>>>>>>", acumulador)
        print("Total =", acumulador)
    elif operacion == 5:
        sys.exit(0)
    else:
        principal()

principal()
